packages/sites/src/drafts/save-published-status.ts

```diff
@@ -20,7 +21,17 @@ export function savePublishedStatus(
   let prms;
 
   if (isSite(item)) {
-    prms = updateSite(siteOrPageModel, allowList, hubRequestOptions);
+    // when saving a draft site, we need to prevent the schemaVersion
+    // from being updated. if not, and the user refreshes the
+    // browser after saving the draft without first publishing,
+    // migration changes can be lost
```

Comment:
I don't quite follow how the changes would be lost, but I trust you.

Reply:
I'm not following this either... @rweber-esri can you explain what at risk of getting lost?

Reply:
This resolves an issue I noticed that can be reproduced by the following steps:
1. Site is at schemaVersion 1.3 in AGO
2. User navigates to /edit, for which schema 1.4 is applied, but not persisted to AGO.
3. User makes local changes & saves draft.
4. When the above method is invoked, we make `typeKeywords` the only thing in the `allowList`, but the `updateSite` method automatically adds `schemaVersion` to the `allowList`, which results in `schemaVersion` being bumped to 1.4 & the typeKeyword representing we have an unpublished draft is added to the item, but none of the migration changes are committed as a result.
5. User refreshes their browser, now the item in AGO has schemaVersion 1.4, so the migration isn't applied.
6. The user saves another draft
7. Migrations changes are lost to the wind

Reply:
Updated the comment to be more accurate wrt the STR in my previous comment